src/codegen.cpp

```diff
@@ -4569,7 +4569,7 @@ static Function *emit_function(jl_lambda_info_t *lam)
                 s->name,                // Variable name
                 topfile,                 // File
                 toplineno == -1 ? 0 : toplineno, // Line (for now, use lineno of the function)
-                julia_type_to_di(varinfo.value.typ, ctx.dbuilder, specsig), // Variable type
+                julia_type_to_di(varinfo.value.typ, ctx.dbuilder, varinfo.value.isboxed), // Variable type
```

Comment:
isboxed isn't really well defined for varinfo.value, but regardless, it hasn't been assigned yet here. i'm not certain that this is what you want here anyways. I think you want to check after emitting the function whether a memloc needed to be created for this variable, or whether it was allocated in `.value` and whether `varinfo.value.typ` is a known type or not.